### PR TITLE
ban some es6 array instance methods that might be widely supported

### PR DESCRIPTION
**Pull request by erwinmombay**
Closes https://github.com/ampproject/amphtml/issues/6539

### CODE REVIEW

**Comment by erwinmombay** [comment]
still trying to figure out how to ban static methods like Array.from and Array.of

**Comment by jridgewell** [comment]
> Array.from and Array.of

Can't we do those in the old presubmit regexes? Very unlikely someone will `var A = Array; A.from(...)`

File: build-system/conformance-config.textproto [hunk 1]
```diff
@@ -4,3 +4,21 @@ requirement: {
   java_class: 'com.google.javascript.jscomp.ConformanceRules$BanThrowOfNonErrorTypes'
   error_message: 'Should not throw a non-Error object.'
 }
+
+requirement: {
+  type: BANNED_PROPERTY_CALL
+  error_message: 'Array.prototype.find is not allowed'
+  value: 'Array.prototype.find'
```
Comment (jridgewell) [inline build-system/conformance-config.textproto]: This isn't failing https://github.com/ampproject/amphtml/issues/6538?

**Comment by erwinmombay** [comment]
@jridgewell hah yeah it isn't running `gulp check-types` since theres no non build system change lol but locally it should be failing #6538 and i won't merge this until that issue is resolved.

yeah we actually have Array.of in the regex

File: build-system/conformance-config.textproto (hunk 1 above)
Comment (erwinmombay) [inline build-system/conformance-config.textproto]: since theres no actual source code change it isn't running `gulp check-types` LOL

Comment (erwinmombay) [inline build-system/conformance-config.textproto]: ill make the change

File: build-system/pr-check.js [hunk 2]
```diff
@@ -80,7 +80,10 @@ function isValidatorWebuiFile(filePath) {
  * @return {boolean}
  */
 function isBuildSystemFile(filePath) {
-  return filePath.startsWith('build-system');
+  return filePath.startsWith('build-system') &&
+      // Exclude textproto from build-system since we want it to trigger
+      // tests and type check.
+      !path.dirname(filePath).endsWith('textproto');
```
Comment (jridgewell) [inline build-system/pr-check.js]: `basename`?

**Comment by erwinmombay** [comment]
@jridgewell should be failing now 
screen shot 2016-12-08 at 11 36 12 am

File: build-system/pr-check.js [hunk 3]
```diff
@@ -80,7 +80,10 @@ function isValidatorWebuiFile(filePath) {
  * @return {boolean}
  */
 function isBuildSystemFile(filePath) {
-  return filePath.startsWith('build-system');
+  return filePath.startsWith('build-system') &&
+      // Exclude textproto from build-system since we want it to trigger
+      // tests and type check.
+      !(path.extname(filePath) == '.textproto');
```
Comment (jridgewell) [inline build-system/pr-check.js]: `!=`. 😛

Comment (erwinmombay) [inline build-system/pr-check.js]: done.

File: build-system/tasks/presubmit-checks.js [hunk 4]
```diff
@@ -413,6 +413,7 @@ var forbiddenTerms = {
 
   // ES6. These are only the most commonly used.
   'Array\\.of': es6polyfill,
+  'Array\\.from': es6polyfill,
```
Comment (erwinmombay) [inline build-system/tasks/presubmit-checks.js]: @jridgewell added `Array.from`

**Comment by jridgewell** [comment]
Ping.

**Pull request merged**